### PR TITLE
Fix of a check on the geometry in the RPCConeBuilder

### PR DESCRIPTION
**Pull request by elfontan**
#### PR description:
Temporary fix of an issue encountered while testing the new CMSSW_12_3_0_pre4 release to run the L1 emulation. 
All details about the issue can be found in this [GitHub issue](https://github.com/cms-sw/cmssw/issues/36786).
Some follow-up is needed with L1 RPC people who agreed for now with the proposed solution of removing the problematic check on the geometry. It is supposed to check if the geometry has a complete ring, however in the case of demo chambers only two sectors are added and the ring is not filled, so that the check fails.

#### PR validation:
Basic tests performed successfully starting from CMSSW_12_3_0_pre4. 
From CMSSW_12_3_0_pre4/src:
> scram b distclean 
> git cms-checkdeps -a -A
> scram b -j 8
> scram b runtests
> scram build code-checks
> scram build code-format
> runTheMatrix.py -l limited -i all --ibeos

### CODE REVIEW

**Comment by missirol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by perrotta** [comment]
please test

File: L1Trigger/RPCTrigger/plugins/RPCConeBuilder.cc [hunk 1]
```diff
@@ -132,7 +134,10 @@ void RPCConeBuilder::buildCones(RPCGeometry const* rpcGeom,
       key -= 100;
     }
 
-    if (key != 2000) {  // Hey 2100 has no counterring
+    // Check if the geometry has a complete ring:
+    // note that in the case of demo chambers, the ring is not filled because only 2 sectors are added.
+    // (3014 and 4014 lack counter-rings)
+    if (key != 2000 && key != 3014 && key != 4014) {  // Hey 2100 has no countering
```
Comment (cvuosalo) [inline L1Trigger/RPCTrigger/plugins/RPCConeBuilder.cc:140]:
```suggestion
    if (key != 2000 && key != 3014 && key != 4014) {  // Key 2100 has no counter-ring
```

**Comment by cvuosalo** [comment]
This solution looks good to me (after comment typo fix).

**Comment by cvuosalo** [comment]
This PR needs to be backported to 12_2.

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
urgent

**Comment by cecilecaillol** [comment]
+l1

**Comment by perrotta** [comment]
+1

**Pull request merged**